code/modules/reagents/chemistry/reagents/water.dm

```diff
@@ -302,14 +303,15 @@
 					M.Jitter(30)
 					if(prob(40))
 						M.emote("scream")
-					M.mind.vampire.adjust_nullification(20, 4)
+					vamp.adjust_nullification(20, 4)
 	return ..() | update_flags
 
 
 /datum/reagent/holywater/reaction_mob(mob/living/M, method=REAGENT_TOUCH, volume)
 	// Vampires have their powers weakened by holy water applied to the skin.
-	if(ishuman(M) && M.mind && M.mind.vampire && !M.mind.vampire.get_ability(/datum/vampire_passive/full))
-		var/mob/living/carbon/human/H=M
+	var/datum/antagonist/vampire/V = M?.mind?.has_antag_datum(/datum/antagonist/vampire)
```

Comment:
```suggestion
	var/datum/antagonist/vampire/V = M.mind?.has_antag_datum(/datum/antagonist/vampire)
```
Same here. If you get a `null` supplied then you should scream in terror since shit fucked up anyway